### DISCUSSION TITLE
Week 2 Status Updates

### DISCUSSION THREAD

**Discussion by roryeiffe** [roryeiffe/Fresh-Results#12] [General]
Respond to this thread with everything you've worked on from 3/25/21-4/1/21. 

What did you work on this week?

What sort of technical blockers are you facing?

What are you going to work on next week?

What sort of new technologies did you learn, if any?

Put any links to commits/other contributions here.

Feel free to add more questions as you think of them!

**Comment by roryeiffe** [comment]
### What did you work on this week?

This week, I worked on the spoiler threshold component for the popup. It is an input box with arrows that increment/decrement the spoiler threshold. The threshold will be used to determine which websites to block, according to how likely they are to spoil things. I also update the style guidelines and start guide to explain how we do CSS styling and how to test the project respectively.

### What sort of technical blockers are you facing?

 I am currently facing an error when trying to build. It is detailed in this [issue](https://github.com/roryeiffe/Spoiler-Block/issues/10
).

### What are you going to work on next week?

Next week, I will look into fixing an error that occurs when trying to build the extension. I will also work on the color picker component that lets users pick the default color for the spoiler blockers. 

### What sort of new technologies did you learn, if any?

I learned how to write React functional components. The key thing that I learned is the useState function. You can define both the values and the function to change the values using useState, and then access those values and change them when applicable in the function. I used this article as a reference: https://reactjs.org/docs/hooks-state.html

### Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/wiki/Style-Guidelines
https://github.com/roryeiffe/Spoiler-Block/issues/10
https://github.com/roryeiffe/Spoiler-Block/commit/78b442e9a334f443870d9779e9120bd82bfaa1c5
https://github.com/roryeiffe/Spoiler-Block/commit/eb689a193c45880d22b871713b8ee5f01e59178b

**Comment by SerenaChen1** [comment]
**What did you work on this week?**
This week, I started working on the "more information" React components for our popup. For each customizable feature, the user should be able to know what exactly that feature does. For example, next to the color selector, there should be a question mark icon the user can hover over to get more details about what the color selector is for.

**What sort of technical blockers are you facing?**
I'm facing the same issue as Rory, where the React code works when we do npm start but not when we build the project.

**What are you going to work on next week?**
Next week, I will likely work on other React components. 

**What sort of new technologies did you learn, if any?**
I spent some time trying to learn React again because I haven't used it in a while.

**Comment by VictoriaSavage526** [comment]
What did you work on this week?
This week, I worked on some user interface tasks and got more up to speed with the chrome extension development process, and also created a more detailed logo. I also merged Rory's Pull Request.

What sort of technical blockers are you facing?
I'm was facing some system issues with installation and git with my system but resolved it.

What are you going to work on next week?
Next week, I will work on enabling the pop-up component to alert the user to spoiler-related content on the page. 

What sort of new technologies did you learn, if any?
I will spend some time reviewing React to ensure that my syntax is proper, as I am not very familiar with it.

**Comment by Sushiman33** [comment]
**What did you work on this week?**
Worked on making an excel sheet with over 100 spoiler related words and a weight associated with them to help with future needs.

**What sort of technical blockers are you facing?**
None at the moment.

**What are you going to work on next week?**
Adding any more words that are needed to prevent spoilers and figure out the best way possible to import the words and their weights and insert them into a dictionary

**What sort of new technologies did you learn, if any?**
Learned some more stuff about JS structure as I am not very familiar with it.

**Comment by BreadGarLick** [comment]
I became more familiar with javascript by messing with simple extensions that I ran locally.
I also helped with brainstorming backend implementation.

**Comment by TheCovenant** [comment]
### What did you work on this week:


I worked with Abdul, Serena, and Rory to create designs based on different use-cases we could think our extension could be used for. It was a very interesting process, because it involved thinking of functionalities in theory, and then bringing them to life by creating mock-ups.

The figma document for our designs:  https://www.figma.com/file/N6RD9WowT8LbuWFit0kE9f/Spoiler-Block

I also chose which part of the UI I desired to work on. I decided to work on the keyword-editor component. A component that should allow the user to create new keywords for the extension to factor in for spoiling. Multiple flows/steps will be required for the component, so it will be a little complex.

### What sort of technical blockers are you facing:

I'm not facing any significant technical blockers at this point. 

### What are you going to work on next week:

As I did not finish the component, I will continue to work on it the following week.

### What sort of new technologies did you learn, if any:

None, but I did solidify my React knowledge a bit more.

### Links to Contributions:
